### ISSUE TITLE
REST ratelimiter (probably) broken

### ISSUE THREAD

**Issue by queer** [Kraigie/nostrum#36]
From [bucket.ex](https://github.com/Kraigie/nostrum/blob/eb71b7b399dc91eb565435d21e6290502df640af/lib/nostrum/api/bucket.ex#L10):

```Elixir
def lookup_bucket(route) do
  route_time = :ets.lookup(:ratelimit_buckets, route)
  global_time = :ets.lookup(:ratelimit_buckets, "GLOBAL")

  Enum.max([route_time, global_time])
end
```

Intuitively, it seems like this ends up doing something to the effect of
```
iex(1)> Enum.max([{"route", 5, 1234567890, 2}, {"GLOBAL", 10, 1234567890, 4}])
{"route", 5, 1234567890, 2}
```
which probably isn't right. 

Also you told me to open this issue https://i.imgur.com/ff4xy3T.png

**Comment by queer** [comment]
More info:

After testing what `bucket.ex` does in an IEx session,

```Elixir
iex(1)> :ets.new(:ratelimit_buckets, [:set, :public, :named_table])
:ratelimit_buckets
iex(2)> :ets.insert(:ratelimit_buckets, {"/route", 5, 1243567890, -1})
true
iex(3)> :ets.insert(:ratelimit_buckets, {"GLOBAL", 10, 1243567890, -1})
true
iex(4)> route_time = :ets.lookup(:ratelimit_buckets, "/route")
[{"/route", 5, 1243567890, -1}]
iex(5)> global_time = :ets.lookup(:ratelimit_buckets, "GLOBAL")
[{"GLOBAL", 10, 1243567890, -1}]
iex(6)> Enum.max([route_time, global_time])
[{"GLOBAL", 10, 1243567890, -1}]
iex(7)> :ets.insert(:ratelimit_buckets, {"/route", 5, 2345678901, -1}) 
true
iex(8)> route_time = :ets.lookup(:ratelimit_buckets, "/route")        
[{"/route", 5, 2345678901, -1}]
iex(9)> global_time = :ets.lookup(:ratelimit_buckets, "GLOBAL")       
[{"GLOBAL", 10, 1243567890, -1}]
iex(10)> Enum.max([route_time, global_time])                           
[{"GLOBAL", 10, 1243567890, -1}]
```

It looks like this is definitely the wrong behaviour, because it's giving us the global ratelimits when the per-route one may be lower - or even empty! - and using that to determine if messages can be sent:

```Elixir
iex(11)> :ets.insert(:ratelimit_buckets, {"/route", 0, 2345678901, -1})
true
iex(12)> route_time = :ets.lookup(:ratelimit_buckets, "/route")        
[{"/route", 0, 2345678901, -1}]
iex(13)> Enum.max([route_time, global_time])                           
[{"GLOBAL", 10, 1243567890, -1}]
```

`Enum.max/2` returns the maximum by [term order](http://elixir-lang.github.io/getting-started/basic-operators.html), according to [the docs](https://hexdocs.pm/elixir/Enum.html#max/2), and since both inputs to `Enum.max/2` are lists containing a single tuple, it doesn't matter what order they're in:

```Elixir
iex(17)> Enum.max([route_time, global_time])                   
[{"GLOBAL", 10, 1243567890, -1}]
iex(18)> Enum.max([global_time, route_time])
[{"GLOBAL", 10, 1243567890, -1}]
```

**Comment by Kraigie** [comment]
Fixed with cb85fe049dbf37ce82935ebf1ee71356b9d02b71, thanks!
